File: CortexSuite/vision/benchmarks/localization/src/c/weightedSample.c
/********************************
Author: Sravanthi Kota Venkata
********************************/

#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "localization.h"

I2D* weightedSample(F2D* w)
{
    I2D *bin;
    F2D *seed;
    int n, i, j;

    n = w->height;
    seed = randWrapper(n, 1);
    bin = iSetArray(n, 1, 0);

    for(i=0; i<n; i++)
    {
        for(j=0; j<n; j++)
        {
            if(asubsref(seed,j) > 0)
                asubsref(bin,j) = asubsref(bin,j) + 1;
        }
        for(j=0; j<n; j++)
            asubsref(seed,j) = asubsref(seed,j) - asubsref(w,i);
    }

    free(seed);
    return bin;
}

File: CortexSuite/vision/benchmarks/localization/src/matlab/weightedSample.m
%function x_gen_id=weightedSample(w)
function bin=weightedSample(w)
n=size(w,1);
seed=randWrapper(n,1);
bin = zeros(n,1);
%x_gen_id=zeros(n,1);
 for i=1:n
     for j=1:n
         if(seed(j,1) > 0)
             bin(j,1) = bin(j,1) + 1;
%             x_gen_id(j,1) = x_gen_id(j,1) + bin(j,1);
         end
     end
%     bin = (seed>0);
%     x_gen_id=x_gen_id+bin;
     seed=seed-w(i,1);
 end
% x_gen_id = bin;
%x_gen=x(:,x_gen_id);

